include/system/dev/vfs.h

```diff
@@ -28,6 +28,14 @@ struct fs_driver {
 	int (*isatty_r)(struct _reent*, void* const);
 	off_t (*lseek_r)(struct _reent*, void* const, off_t, int);
 	int (*ctl)(void* const, const uint32_t, void* const);
+	int (*mkdir_r)(struct _reent*, void* const, mode_t mode);
+	int (*link_r)(struct _reent*, void* const, char*, char*);
+	int (*unlink_r)(struct _reent*, void* const, char*);
+	int (*chdir_r)(struct _reent*, void* const, const char*);
+	int (*chmod_r)(struct _reent*, void* const, const char*);
+	long (*pathconf_r)(struct _reent*, void* const, const char*, int);
+	char* (*getcwd_r)(struct _reent*, void* const, const char*, size_t);
+	int (*stat_r)(struct _reent*, void* const, const char*, struct*);
```

Comment:
whatever implementation we end up with, these signatures are wrong. they should match the functions as they're defined

Reply:
Ok, I'll line up the signatures and everything during the last clean up commit.

Reply:
you might as well do it sooner rather than later. it will make it easier for us to compile it for you